src/one/nio/serial/gen/DelegateGenerator.java

```diff
@@ -433,6 +472,19 @@ private static void generateFromJson(ClassVisitor cv, Class cls, FieldDescriptor
         // Create instance
         mv.visitTypeInsn(NEW, Type.getInternalName(cls));
 
+        // support for calling constructor annotated with @Before
+        for (Class searchClass = cls; searchClass != null; searchClass = searchClass.getSuperclass()) {
```

Comment:
The cycle might look tricky to a seasoned reader (I've spent a minute or two to parse the meaning), but I have no suggestions how to simplify that, unfortunately.

Reply:
Yeah, I am not a fan of constructor call either. Will go trough a few use cases and see if `@Default` would work.